Stop checking if website is on HTTPS for 1-click-unsubscribe [MAILPOET-4859]

samnajian opened the pull request:
## Description

This PR removes the check to see if website is using HTTPS before performing 1-click-unsubscribe.


## QA notes

- From a test website that's using HTTPS,  send a newsletter to yourself.
- Remove HTTPS and convert it to HTTP {domain}/wp-admin/options-general.php
- Check it in and email viewer which supports 1-click unsubscribe
- Unsubscribe using the 1-click-unsubscribe button provided by your email viewer
- Verify the unsubscribe has worked.

## Linked tickets

[MAILPOET-4859]


[MAILPOET-4859]: https://mailpoet.atlassian.net/browse/MAILPOET-4859?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

samnajian committed:
Stop checking if website is on HTTPS for 1-click-unsubscribe

[MAILPOET-4859]

costasovo reviewed (commented):
@samnajian I see this PR contains commits from https://github.com/mailpoet/mailpoet/pull/4587 and one new commit. The new commit doesn't seem to be dependent on the rest from https://github.com/mailpoet/mailpoet/pull/4587. Also, there is a conflict. I would say that it is a mixture of two PRs. Shouldn't this be only about [the last commit](https://github.com/mailpoet/mailpoet/pull/4595/commits/f06f56183e8d7053287fad1fcd950e3514aa3c8f) ?

samnajian commented:
> @samnajian I see this PR contains commits from #4587 and one new commit. The new commit doesn't seem to be dependent on the rest from #4587. Also, there is a conflict. I would say that it is a mixture of two PRs. Shouldn't this be only about [the last commit](https://github.com/mailpoet/mailpoet/pull/4595/commits/f06f56183e8d7053287fad1fcd950e3514aa3c8f) ?

Sorry, it was introduced when I rebased to trunk earlier today, updated.

costasovo reviewed (dismissed):
LGTM! 👍

costasovo commented:
@samnajian I've already approved this PR. But now I noticed it is intended to be merged to branch [MAILPOET-4862-1-click-unsubscribe-should-be-recorded-as-click-stat-as-well](https://github.com/mailpoet/mailpoet/tree/MAILPOET-4862-1-click-unsubscribe-should-be-recorded-as-click-stat-as-well) instead of to trunk. We don't usually do this. Why not merge it to `trunk`?

samnajian commented:
> @samnajian I've already approved this PR. But now I noticed it is intended to be merged to branch [MAILPOET-4862-1-click-unsubscribe-should-be-recorded-as-click-stat-as-well](https://github.com/mailpoet/mailpoet/tree/MAILPOET-4862-1-click-unsubscribe-should-be-recorded-as-click-stat-as-well) instead of to trunk. We don't usually do this. Why not merge it to `trunk`?

Unfortunately the other 1-click branch is merged into this, and I have to either create a new PR for this and cherry pick or merge this PR into https://github.com/mailpoet/mailpoet/pull/4587

Aschepikov reviewed (dismissed):
GJ!

Aschepikov commented:
Hi, there are some conflicts, can you resolve them pls

Aschepikov reviewed (approved):
GJ!
